<?xml version="1.0" encoding="UTF-8"?>

<launch>
	<arg name="slam"/>
	
	<!-- If the user wants SLAM, spawn at the four corners of the map-->
	  <group if="$(arg slam)">

			<!-- Spawn Drone 1 -->
			<group ns="drone1">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/> <!-- -40 -42 -->
					<arg name="y" value="40"/>
					<arg name="drone_name" value="drone1"/>
					<arg name="slam" value="$(arg slam)"/> 
					<arg name="height" value="1.30"/>
				</include>
			</group>
			
			<!-- Spawn Drone 2 -->
			<group ns="drone2">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/> <!-- -42 40 -->
					<arg name="y" value="-40"/>
					<arg name="drone_name" value="drone2"/>
					<arg name="slam" value="$(arg slam)"/> 
					<arg name="height" value="1.16"/>
				</include>
			</group>

			<!-- Spawn Drone 3 -->
			<group ns="drone3">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="38"/>
					<arg name="y" value="40"/>
					<arg name="drone_name" value="drone3"/>
					<arg name="slam" value="$(arg slam)"/> 
					<arg name="height" value="1.02"/>
				</include>
			</group>

			<!-- Spawn Drone 4-->
			<group ns="drone4">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="38"/> <!-- 120 44 -->
					<arg name="y" value="-40"/>
					<arg name="drone_name" value="drone4"/>
					<arg name="slam" value="$(arg slam)"/> 
					<arg name="height" value="0.88"/>
				</include>
			</group>
	</group> 

		<!-- Unless the user is on navigation mode (slam is false)-->
    	<group unless="$(arg slam)">
			<param name="/use_sim_time" value="true"/>
    		<arg name="map_file" default="$(find swarm)/maps/urban_small.yaml"/>
			
			<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
        		<param name="frame_id"  value="map"/>
		    </node>

			<!-- Spawn Drone 1 -->
			<group ns="drone1">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/>
					<arg name="y" value="40"/>
					<arg name="drone_name" value="drone1"/>
					<arg name="slam" value="$(arg slam)"/>
					<arg name="height" value="1.32"/>
 
				</include>
			
				<include file="$(find swarm)/launch/navigation/move_base_nav.launch">
        	   		<arg name="tf_prefix" value="drone1"/>
           			<arg name="odom_frame_id" value="drone1/world"/>
	        	</include>
			</group>
			
			<!-- Spawn Drone 2 -->
			<group ns="drone2">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/>
					<arg name="y" value="22"/>
					<arg name="drone_name" value="drone2"/>
					<arg name="slam" value="$(arg slam)"/> 
					<arg name="height" value="1.21"/>
				</include>

				<include file="$(find swarm)/launch/navigation/move_base_nav.launch">
           			<arg name="tf_prefix" value="drone2"/>
           			<arg name="odom_frame_id" value="drone2/world"/>
		        </include>
			</group>

			<!-- Spawn Drone 3 -->
			<group ns="drone3">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/>
					<arg name="y" value="1"/>
					<arg name="drone_name" value="drone3"/>
					<arg name="slam" value="$(arg slam)"/>
					<arg name="height" value="1.07"/> 
				</include>

				<include file="$(find swarm)/launch/navigation/move_base_nav.launch">
           			<arg name="tf_prefix" value="drone3"/>
        		   <arg name="odom_frame_id" value="drone3/world"/>
		        </include>
			</group>

			<!-- Spawn Drone 4-->
			<group ns="drone4">
				<include file="$(find swarm)/launch/single_drone.launch">
					<arg name="x" value="-42"/>
					<arg name="y" value="-19.6"/>
					<arg name="drone_name" value="drone4"/>
					<arg name="slam" value="$(arg slam)"/>
					<arg name="height" value="0.95"/> 
				</include>

				<include file="$(find swarm)/launch/navigation/move_base_nav.launch">
           		   <arg name="tf_prefix" value="drone4"/>
        		   <arg name="odom_frame_id" value="drone4/world"/>
		        </include>
			</group>

				<include file="$(find swarm)/launch/navigation/amcl.launch">
           			<arg name="tf_prefix" value="drone1"/>
           			<arg name="odom_frame_id" value="drone1/world"/> 
            		<arg name="initial_pose_x" value="-42"/>
        		    <arg name="initial_pose_y" value="40"/>
		        </include>

				<include file="$(find swarm)/launch/navigation/amcl.launch">
            		<arg name="tf_prefix" value="drone2"/>
            		<arg name="odom_frame_id" value="drone2/world"/> 
            		<arg name="initial_pose_x" value="-42"/>
           			<arg name="initial_pose_y" value="22"/>
        		</include>

				<include file="$(find swarm)/launch/navigation/amcl.launch">
           			<arg name="tf_prefix" value="drone3"/>
           			<arg name="odom_frame_id" value="drone3/world"/> 
            		<arg name="initial_pose_x" value="-42"/>
            		<arg name="initial_pose_y" value="1"/>
		        </include>

				<include file="$(find swarm)/launch/navigation/amcl.launch">
            		<arg name="tf_prefix" value="drone4"/>
            		<arg name="odom_frame_id" value="drone4/world"/> 
            		<arg name="initial_pose_x" value="-42"/>
            		<arg name="initial_pose_y" value="-19.6"/>
		        </include>

	 		<!-- Start rviz visualization with preset config -->
 			 <node pkg="rviz" type="rviz" name="rviz" args="-d $(find swarm)/rviz/all_drone_nav.rviz"/>
 	   </group>
</launch>
